### PR TITLE
feat: csv to oscal component definition

### PR DESCRIPTION
**Pull request by degenaro**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [ ] My PR meets testing requirements.
- [ ] All new and existing tests passed.
- [ ] All commits are signed-off.

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: trestle/tasks/csv_to_oscal_cd.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+    def _build_roles(self) -> List[Role]:
+        """Build roles."""
+        value = [
+            Role(id='prepared-by', title='Indicates the organization that created this content.'),
+            Role(id='prepared-for', title='Indicates the organization for which this content was created..'),
+            Role(
+                id='content-approver',
+                title='Indicates the organization responsible for all content represented in the "document".'
+            ),
```
Comment (enikonovad) [inline trestle/tasks/csv_to_oscal_cd.py:241]: Are these hardcoded for every cd?

**Comment by degenaro** [comment]
Thx Ekaterina. None of these look like show stoppers, if you wanted to download this branch as is. I’ll fix these problems tomorrow.

File: trestle/tasks/csv_to_oscal_cd.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+        Args:
+            config_object: Config section associated with the task.
+        """
+        super().__init__(config_object)
+        self.csv_helper = CsvHelper()
+        self._timestamp = datetime.datetime.utcnow().replace(microsecond=0).replace(tzinfo=datetime.timezone.utc
+                                                                                    ).isoformat()
+        self._verbose = False
+        self._ns = 'http://ibm.github.io/compliance-trestle/schemas/oscal/cd'
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/csv_to_oscal_cd.py:68]: Fixed.

File: trestle/tasks/csv_to_oscal_cd.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        return TaskOutcome('simulated-success')
+
+    def execute(self) -> TaskOutcome:
+        """Provide an executed outcome."""
+        try:
+            return self._execute()
+        except Exception:
+            logger.info(traceback.format_exc())
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/csv_to_oscal_cd.py:87]: Fixed.

File: trestle/tasks/csv_helper.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        text2 = '(required) the path of the output directory for synthesized OSCAL .json files.'
+        logger.info(text1 + text2)
+        text1 = '  output-overwrite  = '
+        text2 = '(optional) true [default] or false; replace existing output when true.'
+        logger.info(text1 + text2)
+
+    def configure(self, task: TaskBase) -> bool:
+        """Configure."""
+        if not task._config:
+            logger.warning('config missing')
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/csv_helper.py:106]: Fixed. Return False should be good enuf , caller know what to do in that case.

File: trestle/tasks/csv_helper.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        self.load(csv_file)
+        return True
+
+    def load(self, csv_path: pathlib.Path) -> None:
+        """Load."""
+        with open(csv_path, 'r', newline='') as f:
+            csv_reader = csv.reader(f, delimiter=',', quoting=csv.QUOTE_MINIMAL)
+            for row in csv_reader:
+                self._csv.append(row)
+        self._column.map_head(self._csv[0])
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/csv_helper.py:132]: Fixed.

File: trestle/tasks/csv_to_oscal_cd.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+            responsible_parties=responsible_parties,
+        )
+        component_definition = ComponentDefinition(
+            uuid=str(uuid.uuid4()),
+            metadata=metadata,
+            components=self._get_components(),
+        )
+        # write OSCAL ComponentDefinition to file
+        if self._verbose:
+            logger.info(f'output: {ofile}')
```
[the remark this answers is not in the document]
Comment (degenaro) [reply trestle/tasks/csv_to_oscal_cd.py:132]: I think this is correct as is. `debug` is different from from `verbose`, I think.

File: trestle/tasks/csv_to_oscal_cd.py (hunk 1 above)
Comment (degenaro) [reply trestle/tasks/csv_to_oscal_cd.py:241]: Fixed, by removal as prescribed by Vikas.

**Review by enikonovad — APPROVED** [review]
LGTM - unfortunately, the build is failing, please run `make code-lint` locally and fix some errors, as well as sign-off all commits.

**Pull request merged**